tests/simpleTest/include/simple_test_replica.hpp

```diff
@@ -34,15 +34,20 @@ using namespace bftEngine;
 using namespace bft::communication;
 using namespace std;
 
+namespace bftEngine {
+class ReplicaFactory;
+}
+
 // NOLINTNEXTLINE(misc-definitions-in-headers)
 class SimpleTestReplica {
  private:
   ICommunication *comm = nullptr;
-  bftEngine::IReplica::IReplicaPtr replica = nullptr;
+  std::unique_ptr<IReplica> replica;
```

Comment:
If its a leak, can you show it in sanitizer?
If it not a leak, but only a convertion raw->smart, where is the delete?
otherwisee we risk a double free (crash).

Reply:
Reverted